Dockerfile : Remove ENTRYPOINT for Azure compatibility

themissingcow opened the pull request:
We were using this for devtoolset-6. All manner of attempting to set it in
/etc/propfile.d /root/.bash_profile or /root/.bashrc failed in a variety
of ways. So for now, its source at build time by build.py.

Ideally we'd figure out how to just make this active in the image by default.

themissingcow commented:
@johnhaddon See if this works, I had to give up on trying to get it enabled in the container by default, too many dead ends. Maybe one to come back to later?

johnhaddon commented on build.py:
```diff
@@ -249,19 +257,24 @@ def releaseId() :
 	# have the modules we need to build the docs.
-	buildCommand = "python `which scons` package PACKAGE_FILE={uploadFile} ENV_VARS_TO_IMPORT=PATH DELIGHT_ROOT={delight} ARNOLD_ROOT={arnoldRoot} OPTIONS='' -j {cpus}".format(
-		cpus=multiprocessing.cpu_count(), **formatVariables
+	buildCommand = "{devTools} python `which scons` package PACKAGE_FILE={uploadFile} ENV_VARS_TO_IMPORT=PATH DELIGHT_ROOT={delight} ARNOLD_ROOT={arnoldRoot} OPTIONS='' -j {cpus}".format(
```
Just noting the conclusion from our recent chat : this will break the mac build, and we'd be better to put the `scl enable` stuff up above with the `--entrypoint` stuff in the `containerCommand`.

johnhaddon commented on build.py:
```diff
@@ -99,6 +99,14 @@
 	       "known build platform so that builds are repeatable."
 )
 
+parser.add_argument(
```
Oh, and one other thing just occurred to me. The point of GafferHQ/build is to nail down a single build mechanism we use for all our builds, and we'd both really prefer that all the `scl` stuff was hardcoded into the container itself. So I don't think we really want to add a command line option like this : we want to enforce devtoolset 6.

tomc-cinesite committed:
Build : Remove ENTRYPOINT from Dockerfile

Azure doesn't support docker images that adjust ENTRYPOINT, so we instead
use another trick to ensure that devtoolset-6 is always enabled in this
container.

themissingcow commented:
@johnhaddon We should now have the 🥇
